ocrd_browser/view/images.py

```diff
@@ -76,11 +82,35 @@ def on_size(self, _w: int, h: int, _x: int, _y: int) -> None:
 
     def redraw(self) -> None:
         if self.pages:
-            image: Gtk.Image
-            for image, page in zip_longest(self.image_box.get_children(), self.pages):
-                if page:
-                    thumbnail = pil_scale(page.image, None, self.preview_height - 10)
-                    image.set_from_pixbuf(pil_to_pixbuf(thumbnail))
-                    image.set_tooltip_text(page.id)
-                else:
-                    image.set_from_stock('missing-image', Gtk.IconSize.DIALOG)
+            box: Gtk.Box
+            for box, page in zip_longest(self.image_box.get_children(), self.pages):
+                existing_images = {child.get_name():
+                                   child for child in box.get_children()}
+                for i, img in enumerate(page.images if page else [None]):
+                    name = 'image_{}'.format(i)
+                    image: Gtk.Image
+                    image = existing_images.pop(name, None)
+                    if not image:
+                        image = Gtk.Image(name=name, visible=True,
+                                          icon_name='gtk-missing-image',
+                                          icon_size=Gtk.IconSize.DIALOG)
+                        box.add(image)
+                    if img:
+                        thumbnail = pil_scale(img, None, self.preview_height - 10)
+                        image.set_from_pixbuf(pil_to_pixbuf(thumbnail))
+                        img_file = page.image_files[i]
+                        if img_file == page.file:
+                            image.set_tooltip_text(page.id)
+                        else:
+                            # get segment ID for AlternativeImage as tooltip
+                            img_id = page.pc_gts.gds_elementtree_node_.xpath(
```

Comment:
This line raises:
``` 
AttributeError: 'NoneType' object has no attribute 'xpath'
``` 
page.pc_gts.gds_elementtree_node_ is None

Reply:
> This line raises:
> 
> ```
> AttributeError: 'NoneType' object has no attribute 'xpath'
> ```
> 
> page.pc_gts.gds_elementtree_node_ is None

Hmm, I see. And under which circumstances does this arise (what kind of pageId in what kind of fileGrp is selected)?

@kba what's the reliable way of XPath querying after ocrd_page.parse?

Reply:
> @kba what's the reliable way of XPath querying after ocrd_page.parse?

PcGtsType *should* have `self.gds_elementtree_node_` set, I don't understand why it isn't - did it work for you locally?

@hnesk are you on the latest core version 2.18.0?

One rather extreme solution is what we do for `get_AllAlternativeImagePaths`:

```python
    sio = StringIO()
    self.export(
            outfile=sio,
            level=0,
            name_='PcGts',
            namespaceprefix_='pc:',
            namespacedef_='xmlns:pc="%s" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="%s %s/pagecontent.xsd"' % (
                NAMESPACES['page'],
                NAMESPACES['page'],
                NAMESPACES['page']
            ))
    doc = parsexmlstring_(sio.getvalue())  # pylint: disable=undefined-variable
    # shortcut
    if page and region and line and word and glyph:
        ret += doc.xpath('//page:AlternativeImage/@filename', namespaces=NAMESPACES)
    # etc.
```

Reply:
> PcGtsType _should_ have `self.gds_elementtree_node_` set, I don't understand why it isn't - did it work for you locally?

Yes, it worked perfectly.

> One rather extreme solution is what we do for `get_AllAlternativeImagePaths`:

Yes, I saw it and thought about doing something similar, but then found `gds_elementtree_node_` and was happy :-)

It's probably best to debug the mets.xml that caused this directly.

Reply:
It happened for me with an "autogenerated" PAGE-XML from ocrd_modelfactory.page_from_image().
This was an old workspace, where PAGE-XML and images did not reside in the same file_grp. 
I will implement a simple `gds_elementtree_node_ is not None` check

Reply:
> It happened for me with an "autogenerated" PAGE-XML from ocrd_modelfactory.page_from_image().
> This was an old workspace, where PAGE-XML and images did not reside in the same file_grp.

I see. I had tested a little on such a workspace myself, but did not detect this. :confused:

> I will implement a simple `gds_elementtree_node_ is not None` check

No, wait! I know what happened: I did check – but only clicked the **first image**! My check `page.image_files[i] == page.file`  works there, but does not for `i>0`. It's better to fix that criterion than attempting the XPath. I'll fix this myself.

(Also, I have an image zoom feature with ctrl + scroll wheel and button nearly finished. It's based on the same branch.)

Reply:
> I'll fix this myself.
> 
> (Also, I have an image zoom feature with ctrl + scroll wheel and button nearly finished. It's based on the same branch.)

Done!